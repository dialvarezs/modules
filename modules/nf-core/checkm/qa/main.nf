process CHECKM_QA {
    tag "${meta.id}"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/checkm-genome:1.2.3--pyhdfd78af_1' :
        'biocontainers/checkm-genome:1.2.3--pyhdfd78af_1' }"

    input:
    tuple val(meta), path(analysis_dir), path(marker_file), path(coverage_file)
    path exclude_marker_file
    val tsv_output

    output:
    tuple val(meta), path("${prefix}.txt")  , optional: true, emit: output
    tuple val(meta), path("${prefix}.fasta"), optional: true, emit: fasta
    path "versions.yml"                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args       = task.ext.args ?: ''
    prefix         = task.ext.prefix ?: "${meta.id}"
    suffix         = task.ext.args?.matches(".*-o 9.*|.*--out_format 9.*") ? "fasta" : "txt"
    def tsv_format = tsv_output                                            ? "--tab_table" : ""
    def coverage   = coverage_file.isFile()                                ? "--coverage_file ${coverage_file}"  : ""
    def exclude    = exclude_marker_file && exclude_marker_file.isFile()   ? "--exclude_markers ${exclude_marker_file}" : ""
    """
    checkm \\
        qa \\
        --threads ${task.cpus} \\
        --file ${prefix}.${suffix} \\
        ${marker_file} \\
        ${analysis_dir} \\
        ${coverage} \\
        ${tsv_format} \\
        ${exclude} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        checkm: \$( checkm 2>&1 | grep '...:::' | sed 's/.*CheckM v//;s/ .*//' )
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.txt ${prefix}.fasta

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        checkm: \$( checkm 2>&1 | grep '...:::' | sed 's/.*CheckM v//;s/ .*//' )
    END_VERSIONS
    """
}
