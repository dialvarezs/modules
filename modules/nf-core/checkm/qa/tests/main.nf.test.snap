{
    "Test CheckM QA": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.txt:md5,645f4282569afb4b171396732b2d2582"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,7a0683a78cbf54a6a69ee64055c584a6"
                ],
                "fasta": [
                    
                ],
                "output": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.txt:md5,645f4282569afb4b171396732b2d2582"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7a0683a78cbf54a6a69ee64055c584a6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-05T02:56:27.941219756"
    }
}